test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/ContentResultTest.cs

```diff
@@ -73,45 +79,60 @@ public class ContentResultTest
             Assert.Equal("text/plain; charset=utf-7", httpContext.Response.ContentType);
         }
 
-        [Fact]
-        public async Task ContentResult_Response_NullContent_SetsContentTypeAndEncoding()
+        public static TheoryData<MediaTypeHeaderValue, string, byte[]> ContentResultContentTypeData
         {
-            // Arrange
-            var contentResult = new ContentResult
+            get
             {
-                Content = null,
-                ContentType = "application/json",
-                ContentEncoding = Encoding.UTF8
-            };
-            var httpContext = GetHttpContext();
-            var actionContext = GetActionContext(httpContext);
-
-            // Act
-            await contentResult.ExecuteResultAsync(actionContext);
-
-            // Assert
-            Assert.Equal("application/json; charset=utf-8", httpContext.Response.ContentType);
+                return new TheoryData<MediaTypeHeaderValue, string, byte[]>
+                {
+                    {
+                        null,
+                        "text/plain; charset=utf-8",
+                        new byte[] { 97, 98, 99, 100 } //utf-8 without BOM
+                    },
+                    {
+                        new MediaTypeHeaderValue("text/foo"),
+                        "text/foo; charset=utf-8",
+                        new byte[] { 97, 98, 99, 100 } //utf-8 without BOM
+                    },
+                    {
+                        MediaTypeHeaderValue.Parse("text/foo;p1=p1-value"),
+                        "text/foo; p1=p1-value; charset=utf-8",
+                        new byte[] { 97, 98, 99, 100 } //utf-8 without BOM
+                    },
+                    {
+                        new MediaTypeHeaderValue("text/foo") { Encoding = Encoding.ASCII },
+                        "text/foo; charset=us-ascii",
+                        new byte[] { 97, 98, 99, 100 }
+                    }
+                };
+            }
         }
 
-        [Fact]
-        public async Task ContentResult_Response_BadContentType_ThrowsFormatException()
+        [Theory]
+        [MemberData(nameof(ContentResultContentTypeData))]
+        public async Task ContentResult_SetContentTypeAndEncoding(
+            MediaTypeHeaderValue contentType,
+            string expectedContentType,
+            byte[] expectedContentData)
         {
             // Arrange
             var contentResult = new ContentResult
             {
-                Content = "Test Content",
-                ContentType = "some-type",
-                ContentEncoding = null
+                Content = "abcd",
```

Comment:
Maybe have a string with some Utf8 chars so we can differentiate between ASCII and UTF8 in the test